araproc/framework/map_utilities.py

```diff
@@ -1,74 +1,309 @@
+        """
+        # Constants
+        self.FEET_TO_METER = 0.3048  # Conversion factor from feet to meter
+        self.SUR_TO_GLOB = np.array([22400, 53900, -36.02])  # Survey to global offset in feet
+
+        self.station_id = station_id
+        # Initialize the Geometry Tool
+        self.geomTool = ROOT.AraGeomTool.Instance()
+
+        # Station information
+        self.st_info = self.geomTool.getStationInfo(self.station_id)
+
+
+    def get_station_center(self, excluded_channels=None):
         """
-        Little utility helper function
+        Calculates the average x, y, and z coordinates of antennas for a given station.
 
         Parameters
         ----------
-        the_map : ROOT.TH2D
-            A 2D map you want the peak found
-        
-        
+        excluded_channels : list
+           list of channels (antennas) to exclude.
+
         Returns
-        peak_corr : float
-            The peak in the correlation  map
-        peak_phi : int
-            The theta value of the peak of the correlation map
-        peak_phi : int
-            The phi value of the peak of the correlation map
+        -------
+        av_ant_position:tuple
+           station center (x, y, z) for given station
         """
-        
-        if the_map is None:
-            raise Exception("No TH2D map was passed")
-    
-        _peakZ = ctypes.c_int()
-        _peakTheta = ctypes.c_int()
-        _peakPhi = ctypes.c_int()
-        the_map.GetMaximumBin(_peakPhi, _peakTheta, _peakZ)
 
-        peak_corr = the_map.GetMaximum()
-        peak_phi = the_map.GetXaxis().GetBinCenter(_peakPhi.value)
-        peak_theta = the_map.GetYaxis().GetBinCenter(_peakTheta.value)
-        return peak_corr, peak_phi, peak_theta
+        if excluded_channels is None:
+            excluded_channels = []
 
-def calculate_avg_antenna_xyz(station_id, num_channels):
-    """
-    Calculates the average x, y, and z coordinates of antennas for a given station.
+        antenna_coordinates = [[] for _ in range(3)]  # x, y, z
+        total_antennas = constants.num_rf_channels
 
-    Parameters
-    ----------
-    station_id : int
-        ID of the station from which to retrieve antenna information.
-    num_channels : int
-        Number of channels (antennas) at the station.
+        for antenna_index in range(total_antennas):
+            if antenna_index in excluded_channels:
+                continue
+
+            antenna_info = self.st_info.getAntennaInfo(antenna_index)
+            for axis in range(3):
+                antenna_coordinates[axis].append(antenna_info.antLocation[axis])
+        av_ant_position = tuple(np.average(coords) for coords in antenna_coordinates)
+        return av_ant_position
+
+
+    def get_local_CP(self, list_of_cal_pulser_indices):
+        """
+        Parameters
+        ----------
+        list_of_cal_pulser_indices: list
+            The indices of calpulser [0,1,2,3] Note: 0 and 2 for HPols and 1 and 3 are VPols
```

Comment:
And [0,1] are on the same string I guess?

Reply:
Yes, 0 and 1 are Hpol and Vpol in one and 2 and 3 are Hpol and Vpol in another string